src/commands/data/export/bulk.ts

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as fs from 'node:fs';
+import { platform } from 'node:os';
+import { SfCommand, Flags } from '@salesforce/sf-plugins-core';
+import { Messages, Org } from '@salesforce/core';
+import { MultiStageOutput } from '@oclif/multi-stage-output';
+import terminalLink from 'terminal-link';
+import { QueryJobInfoV2, QueryJobV2 } from '@jsforce/jsforce-node/lib/api/bulk2.js';
+import { Duration } from '@salesforce/kit';
+import { BulkExportRequestCache } from '../../../bulkDataRequestCache.js';
+import { exportRecords } from '../../../bulkUtils.js';
+
+Messages.importMessagesDirectoryFromMetaUrl(import.meta.url);
+const messages = Messages.loadMessages('@salesforce/plugin-data', 'data.export.bulk');
+
+export type DataExportBulkResult = {
+  jobId?: string;
+  totalSize?: number;
+  filePath: string;
```

Comment:
async export returns `filePath` and `jobId`
sync export returns `filePath` and `totalSize`